Common/cpp/LayoutAnimations/LayoutAnimationsManager.cpp

```diff
@@ -73,4 +94,22 @@ void LayoutAnimationsManager::startLayoutAnimation(
       config->getJSValue(rt));
 }
 
+/*
+  The top screen on the stack triggers the animation, so we need to find
+  the sibling view registered in the past. That's why we look backward.
+
+  We need to find view registered in the same transition group (with the
+  same transition tag) which has been added to that group directly before
+  the one that we provide as an argument.
```

Comment:
I'd rather structure similar comments in a form of "this method does that" rather than "we need so we do"

Reply:
Resolved by: https://github.com/software-mansion/react-native-reanimated/pull/4014/commits/83c74794453818abe6da4894412860e9811e5aef